src/Native/LdaNative/model_block.cpp

```diff
@@ -113,20 +124,22 @@ namespace lda
         dict_ = new WordEntry[num_vocabs_];
         for (int i = 0; i < num_vocabs_; ++i)
         {
-            // This warning is a false positive. Supressing it similar to the existing one on Line 140 below.
-#pragma warning(suppress: 6386)
             dict_[i].is_dense_ = 0;
             dict_[i].is_alias_dense_ = 0;
         }
 
-        mem_block_size_ = mem_block_size;
+		if (mem_block_size < 0 || static_cast<uint64_t>(mem_block_size) > numeric_limits<size_t>::max())
+			throw bad_alloc();
+        mem_block_size_ = static_cast<size_t>(mem_block_size);
         mem_block_ = new int32_t[mem_block_size_]();   // NOTE : force to initialize the values to be zero
 
-        alias_mem_block_size_ = alias_mem_block_size;
+		if (mem_block_size < 0 || static_cast<uint64_t>(alias_mem_block_size) > numeric_limits<size_t>::max())
+			throw bad_alloc();
+        alias_mem_block_size_ = static_cast<size_t>(alias_mem_block_size);
         alias_mem_block_ = new int32_t[alias_mem_block_size_]();    //NOTE: force to initialize the values to be zero
 
-        std::cout << "mem_block_size = " << mem_block_size_ * 4 << std::endl;
-        std::cout << "alias_mem_block_size = " << alias_mem_block_size_ * 4 << std::endl;
+        cout << "mem_block_size = " << mem_block_size_ * 4 << endl;
```

Comment:
>4 [](start = 57, length = 1)

sizeof(float)?

Reply:
you mean  sizeof(int32_t)? (those are arrays of int32_t )

---
In reply to: [276830689](https://github.com/dotnet/machinelearning/pull/3370#discussion_r276830689) [](ancestors = 276830689)